### ISSUE TITLE
[MCHECKSTYLE-323] usage of checkstyle 7.0 (jdk8 is required)

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-checkstyle-plugin#615]
**[Roman Ivanov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=romani)** opened **[MCHECKSTYLE-323](https://issues.apache.org/jira/browse/MCHECKSTYLE-323?redirect=false)** and commented

Checkstyle 7.0 requires jdk8 for runtime.

It will be good to have special version of maven plugin that is using it, as good milestone.

---

**Issue Links:**
- [MCHECKSTYLE-366](https://issues.apache.org/jira/browse/MCHECKSTYLE-366) Upgrade checkstyle to a more recent version
 (_**"duplicates"**_)

2 votes, 5 watchers

**Comment by jira-importer** [comment]
**[Robert Metzger](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rmetzger)** commented

You can freely choose the checkstyle version you want to use with the plugin: https://maven.apache.org/plugins/maven-checkstyle-plugin/examples/upgrading-checkstyle.html

**Comment by jira-importer** [comment]
**[Roman Ivanov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=romani)** commented

I know this feature and heavily use it.

By this issue I mean: as plugin switch to jdk8 as minimum jdk, it would be better to use checkstyle 7.0 as version by default.

Checkstyle will soon switch to version 8.X there will be a lot of API changes, it will be another good milestone for plugin.

**Comment by jira-importer** [comment]
**[Roman Ivanov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=romani)** commented

One more point to make next version to be based on 7.0 :
Checkstyle is used in many places: maven, IDEs, Sonar like application.
Unfortunately user have to keep checkstyle version in the same version in all systems. So if you give user 7.0 by default, user can use any 7.X he wants or other systems demands.

https://maven.apache.org/plugins/maven-checkstyle-plugin/history.html

**Comment by jira-importer** [comment]
**[Vincent Privat](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=don-vip)** commented

We have a project com.xxx.yyy that requires JAVA_HOME to be set to Java 7 and still want to use the latest version of Checkstyle available with Eclipse (7.3 at this time).
In order to do it we tried this configuration to use a jdk8 toolchain:

```xml
<plugin>
     <groupId>org.apache.maven.plugins</groupId>
     <artifactId>maven-checkstyle-plugin</artifactId>
     <version>2.17</version>
     <dependencies>
              <dependency>
                       <groupId>com.xxx.yyy</groupId>
                       <artifactId>yyy-checkstyle</artifactId>
                       <version>2.7.3</version>
              </dependency>
              <dependency>
                       <groupId>com.puppycrawl.tools</groupId>
                       <artifactId>checkstyle</artifactId>
                       <version>7.3</version>
              </dependency>
     </dependencies>
     <configuration>
              <configLocation>conf/checkstyle/checkstyle_config_maven.xml</configLocation>
              <packageNamesLocation>checkstyle_packages.xml</packageNamesLocation>
     <suppressionsLocation>conf/checkstyle/src/main/resources/checkstyle_suppressions.xml</suppressionsLocation>
              <toolchains>
                       <jdk><version>[1.8,)</version></jdk>
              </toolchains>
     </configuration>
</plugin>
```

Buut it doesn't seem to work. I don't know if my configuration is wrong, or if the plugin is not toolchain-aware?
